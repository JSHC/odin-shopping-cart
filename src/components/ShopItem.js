const ShopItem = (props) => {
    const { id, name, price } = props;
    return (
        <div>
            <span className="shop-item-name">{name}</span>
            <span className="shop-item-price">{price}</span>
        </div>
    )
}

export default ShopItem;