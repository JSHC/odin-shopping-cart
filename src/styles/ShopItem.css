.shop-item {
    display: grid;
    grid-template-rows: 1fr auto auto auto;
    border: 1px solid #ddd;
}

.shop-item div {
    padding: 1rem;
    box-sizing: border-box;
}

.shop-item-image {
    width: 100%;
    height: 300px;
    background-size: cover;
    background-position: center center;
    background-clip: content-box;
}

.shop-item-details {
    grid-row: 2;
}

.shop-item-details>p {
    margin: 0;
}

.shop-item .shop-item-details {
    padding-top: 0;
}

.shop-item-controls {
    grid-row: 4;
    display: grid;
    grid-template-columns: repeat(3, 1fr);
    border-top: 1px solid #ddd;
}

.shop-item-quantity-label {
    grid-row: 1;
    grid-column: 1 / -1;
}

.shop-item-price::before {
    content: "$";
}

.add-to-cart-button {
    grid-column: 1 / -1;
    padding: 1rem 0;
    border: 2px solid #888;
    background-color: #fff;
    cursor: pointer;
    font-weight: bold;
    margin-top: 1rem;
}

#shop-item-quantity {
    padding: 1rem 0.6rem;
    text-align: center;
    font-size: 16px;
}