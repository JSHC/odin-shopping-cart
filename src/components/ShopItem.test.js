import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import ShopItem from './ShopItem';

describe('Home component', () => {
    it('renders', () => {
        const { container } = render(<ShopItem />);
        expect(container).toMatchSnapshot();
    })
    it('renders item', () => {
        const item1 = {id: 1, name: 'Item1', price: 10}
        render(<ShopItem {...item1}/>);
        screen.getByText('Item1');
        screen.getByText('10');
    })
})