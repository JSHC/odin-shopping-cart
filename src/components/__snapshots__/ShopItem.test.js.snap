// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home component renders 1`] = `
<div>
  <div>
    <span
      class="shop-item-name"
    />
    <span
      class="shop-item-price"
    />
  </div>
</div>
`;
